### ISSUE TITLE
I get a strange loss curve, Has anyone encountered the same situation?

### ISSUE THREAD

**Issue by PeijiYang** [TensorSpeech/TensorFlowTTS#510]
![image](https://user-images.githubusercontent.com/17819608/109446024-d818c480-7a7b-11eb-9fbf-ae4ab4ab3d92.png)

Loss was interrupted in some steps, and then began to fall again. Is the size of my dataset too small?

**Comment by dathudeptrai** [comment]
@PeijiYang a bit overfitted :))

**Comment by PeijiYang** [comment]
200k step alignment figure：
![31_alignment](https://user-images.githubusercontent.com/17819608/109452617-1ec1eb00-7a8b-11eb-9e19-19c22d869a7e.png)
![30_alignment](https://user-images.githubusercontent.com/17819608/109452620-1ff31800-7a8b-11eb-85cd-0777c3115a44.png)
![29_alignment](https://user-images.githubusercontent.com/17819608/109452623-21244500-7a8b-11eb-8669-ed58c9b7472c.png)
100k:
![31_alignment (1)](https://user-images.githubusercontent.com/17819608/109452681-47e27b80-7a8b-11eb-8fec-505e94af46f2.png)
![30_alignment (1)](https://user-images.githubusercontent.com/17819608/109452684-4a44d580-7a8b-11eb-9573-d472246bc180.png)
![29_alignment (1)](https://user-images.githubusercontent.com/17819608/109452688-4c0e9900-7a8b-11eb-94f1-a1143c4ebe0d.png)

With obvious shadows

**Comment by dathudeptrai** [comment]
@PeijiYang not really good but may enough for fastspeech training. is ur data small ?

**Comment by PeijiYang** [comment]
Yep, only 2 hours, 1400+ parallel data.  It is strange why there are two interruptions in the loss curve of training. Can FastSpeech use less data for better results than Tacotron2?

**Comment by dathudeptrai** [comment]
> @dathudeptrai Yep, only 2 hours, 1400+ parallel data. It is strange why there are two interruptions in the loss curve of training. Can FastSpeech use less data for better results than Tacotron2?

@PeijiYang could be :D . let try :D.

**Comment by ZDisket** [comment]
@PeijiYang
> only 2 hours

You can get a large public dataset in the same language you're using, train that first, then use your data to fine tune said model. Then you can get good results with Tacotron2 with as little as 30 minutes of audio.

**Comment by PeijiYang** [comment]
> > @dathudeptrai Yep, only 2 hours, 1400+ parallel data. It is strange why there are two interruptions in the loss curve of training. Can FastSpeech use less data for better results than Tacotron2?
> 
> @PeijiYang could be :D . let try :D.

Thank you, I will try it

**Comment by PeijiYang** [comment]
> @PeijiYang
> 
> > only 2 hours
> 
> You can get a large public dataset in the same language you're using, train that first, then use your data to fine tune said model. Then you can get good results with Tacotron2 with as little as 30 minutes of audio.

That cool, I have one same language public dataset with about 10hours, 10000 parallel data. And it is single-speaker dataset. May it is enough to train a pretrained model for Tacotron2?

**Comment by ZDisket** [comment]
> That cool, I have one same language public dataset with about 10hours, 10000 parallel data. And it is single-speaker dataset. May it is enough to train a pretrained model for Tacotron2?

Yes. Although if you don't value the non deterministic (one input, infinite possibilities of results) nature of Tacotron2 then I'd recommend using FastSpeech2.

**Comment by PeijiYang** [comment]
@dathudeptrai @ZDisket FastSpeech2 training seems a bit slow. I checked the GPU occupancy througn nvidia-smi, and I found that Memory-Usage is 10G, but the GPU-Util is always 0%. I don't know if this is a normal situation for training FastSpeech2. Can you tell me how many hours it takes to train FastSpeech2 with 150k steps?

**Comment by ZDisket** [comment]
@PeijiYang A FastSpeech2 that finished training a day ago took about 2 days to get to 150k steps, but that is with batch_size 32 and 4 gradient accumulation steps on a single V100.

**Comment by PeijiYang** [comment]
> @PeijiYang A FastSpeech2 that finished training a day ago took about 2 days to get to 150k steps, but that is with batch_size 32 and 4 gradient accumulation steps on a single V100.

I train it on two V100 with batch_size 64. Do you have stiuation that GPU-Utils is always 0%？i guess that some data processing modules were interspersed during the training process.

**Comment by ZDisket** [comment]
@PeijiYang I don't usually pay attention to GPU utilization unless I notice something wrong, so far I haven't had any of those situations.

**Comment by dathudeptrai** [comment]
> > @PeijiYang A FastSpeech2 that finished training a day ago took about 2 days to get to 150k steps, but that is with batch_size 32 and 4 gradient accumulation steps on a single V100.
> 
> I train it on two V100 with batch_size 64. Do you have stiuation that GPU-Utils is always 0%？i guess that some data processing modules were interspersed during the training process.

 GPU-Util is 0% means at that time it is reading dataset (very sure :)), you should check ur CPU speed.)

**Comment by PeijiYang** [comment]
The speed returned to normal after about 1000 steps, GPU Utils is high now. Thank you.

**Comment by PeijiYang** [comment]
@dathudeptrai @ZDisket It's surprise, fastspeech2 is much better than tacotron2 on my 2 hours datasets. But it is still overfitting to my training data set, and the effect is not very satisfactory. Maybe I need to try finetune fastspeech2 or tacotron2, is it a good way?
